### PR TITLE
[TensorV2] Multiplication support

### PR DESCRIPTION
**Pull request by djeong20**
This PR adds support for performing the multiplication operation on two tensors.

**Changes proposed in this PR:**
- TensorV2 includes member functions to perform tensor multiplication.
- FloatTensor and HalfTensor take TensorV2 as input/output to perform multiplication.

**Self-evaluation:**
1. Build test: [X]Passed [ ]Failed [ ]Skipped
2. Run test:   [X]Passed [ ]Failed [ ]Skipped

### CODE REVIEW

File: nntrainer/tensor/float_tensor.cpp [hunk 1]
```diff
@@ -325,6 +325,56 @@ void FloatTensor::copyData(const TensorV2 &from) {
...
+  return out;
+}
+
+TensorV2 &FloatTensor::multiply(TensorV2 const &m, TensorV2 &output,
+                                const float beta) const {
+  auto f = [&](const BroadcastInfoV2 &e, const float *buf, const float *m_buf,
+               float *out_buf) {
+    if (e.strides[3] == 1 && output.getStrides()[3] == 1 && strides[3] == 1 &&
+        beta == 0.0) {
+      std::transform(buf, buf + e.buffer_size, m_buf, out_buf,
```
Comment (jijoongmoon) [inline nntrainer/tensor/float_tensor.cpp:349]: Even though we use this std::transform, we haven't tested it with other libs or implementations. Not in this PR, but we have to evaluate it.

File: nntrainer/tensor/float_tensor.cpp [hunk 2]
```diff
@@ -325,6 +325,56 @@ void FloatTensor::copyData(const TensorV2 &from) {
...
+                     std::multiplies<float>());
+    } else {
+      for (unsigned int i = 0; i < e.buffer_size; ++i) {
+        *out_buf = *buf * *m_buf + beta * *out_buf;
+        buf += strides[3];
+        m_buf += e.strides[3];
+        out_buf += output.getStrides()[3];
```
Comment (jijoongmoon) [inline nntrainer/tensor/float_tensor.cpp:356]: If it is not continuous, we still can apply the multi-threading with something like omp. Not this PR, but we can also evaluate it.

**Review by jijoongmoon — APPROVED** [review]
LGTM except for some comments.

File: nntrainer/tensor/float_tensor.cpp (hunk 1 above)
Comment (djeong20) [reply nntrainer/tensor/float_tensor.cpp:349]: gotcha. I'll add some testing to verify.

File: nntrainer/tensor/float_tensor.cpp (hunk 2 above)
Comment (djeong20) [reply nntrainer/tensor/float_tensor.cpp:356]: agree. We can launch a thread using a function pointer or function objects in multithreading support.

**Review by SeoHyungjun — APPROVED** [review]
LGTM!

**Pull request merged**